Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -287,18 +296,37 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
     )
     .takeWhen(didTapToAddNewCard)
     .switchMap { (project, _) -> SignalProducer<Signal<PaymentSheetSetupData, ErrorEnvelope>.Event, Never> in
-      AppEnvironment.current.apiService
-        .createStripeSetupIntent(input: CreateSetupIntentInput(projectId: project.graphID))
+
+      let usePaymentIntent = featurePostCampaignPledgeEnabled() && project.isInPostCampaignPledgingPhase
+      let useSetupIntent = !usePaymentIntent
+      let clientSecretSignal: SignalProducer<String, ErrorEnvelope>
+
+      if useSetupIntent {
+        clientSecretSignal = AppEnvironment.current.apiService
+          .createStripeSetupIntent(input: CreateSetupIntentInput(projectId: project.graphID))
+          .map { $0.clientSecret }
+      } else {
+        // TODO: real amt, real digital marketing attribution
+        clientSecretSignal = AppEnvironment.current.apiService
+          .createPaymentIntentInput(input: CreatePaymentIntentInput(
+            projectId: project.graphID,
+            amountDollars: "10.00",
```

Comment:
This is the important ``// TODO`` here - we need the actual pledge amount in this view model.